infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -363,6 +372,10 @@ private void applyShadowStream(UUID streamId, long snapshot) {
             smrEntries.size(), numBatches);
     }
 
+    private boolean maxEntriesLimitReached(UUID streamId, List<SMREntry> buffer) {
+        return (streamId.equals(PROTOBUF_TABLE_ID) && buffer.size() == config.getMaxSnapshotEntriesApplied());
```

Comment:
Can you remind me why we only do it for protoBuf table?

Reply:
In one particular bug due to a number of restarts the tables got re-opened consequtively in old and new versions causing the protobuf descriptor entries to flip back and forth increasing the number of updates to the same entry during upgrade